src/error.rs

```diff
@@ -21,6 +33,28 @@ pub type DaftResult<T> = std::result::Result<T, DaftError>;
 impl Display for DaftError {
     // `f` is a buffer, and this method must write the formatted string into it
     fn fmt(&self, f: &mut Formatter) -> Result {
-        write!(f, "{self:?}")
+        // Override this method if the error propagated should have custom display for better user-level error messages
+        match self {
+            Self::ExprResolveTypeError {
+                expectation,
+                expr_repr,
+                child_fields_to_expr,
+            } => {
+                writeln!(
+                    f,
+                    "Expects {expectation}, but failed type resolution: {expr_repr}",
+                )?;
+                writeln!(f, "where expression arguments are:")?;
+                for (field, expr) in child_fields_to_expr.iter() {
```

Comment:
what you had earlier was fine, I was saying to use`.iter().for_each(...)` instead of `.iter().map(...)` if you don't need the result

Reply:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each

Reply:
Yes, I just thought this was a little easier to read than the `for_each`